lib/msf/core/opt.rb

```diff
@@ -41,7 +41,7 @@ def self.Proxies(default=nil, required=false, desc="A proxy chain of format type
 
     # @return [OptAddressRange]
     def self.RHOSTS(default=nil, required=true, desc="The target host(s), range CIDR identifier, or hosts file with syntax 'file:<path>'")
-      Msf::OptAddressRange.new('RHOSTS', [ required, desc, default ])
+      Msf::OptAddressRange.new('RHOSTS', [ required, desc, default ], aliases: [ 'RHOST' ])
```

Comment:
It _seemed_ like it was an intentionally made decision at the time to not add RHOST as an alias:
https://github.com/rapid7/metasploit-framework/pull/10096

But it's interesting that it did work under certain circumstances anyway. I believe it was because the aliasing support also happens in the datastore as well as the option itself:

https://github.com/rapid7/metasploit-framework/blob/66de4deabf9a403b42cc70f0e94f7d586de3c4e7/lib/msf/core/data_store.rb#L143-L154

So if you register `RHOSTS` as an option, for a module that's already registered RHOST, the datastore itself is mutated to have the aliases setup. Even if no option supports the alias by the time the module runs